library/src/hipfft.cpp

```diff
@@ -317,7 +317,9 @@ hipfftResult hipfftMakePlan_internal(hipfftHandle            plan,
                                         dry_run));
         break;
     default:
-        assert(false);
+        std::stringstring ss;
+        ss << "hipfftType: " << type << " not recognized!");
+        throw std::runtime_exception(ss.str());
```

Comment:
Throw exception from shared library is not a good practice for code portability.
Does "return HIPFFT_PARSE_ERROR" work for you?

Reply:
Yes, this totally works (and is more in keeping with the error checking status anyways)

Reply:
updated